src/mame/misc/astrcorp.cpp

```diff
@@ -281,128 +437,191 @@ void astrocorp_state::draw_sprites(bitmap_ind16 &bitmap, const rectangle &clipre
 
 u32 astrocorp_state::screen_update(screen_device &screen, bitmap_ind16 &bitmap, const rectangle &cliprect)
 {
-	if (m_screen_enable & 1)
+	if (BIT(m_screen_enable, 0))
 		copybitmap(bitmap, m_bitmap, 0,0,0,0, cliprect);
 	else
 		bitmap.fill(m_palette->black_pen(), cliprect);
 
 	return 0;
 }
 
-
 /***************************************************************************
                                 Memory Maps
 ***************************************************************************/
 
 void astrocorp_state::draw_sprites_w(offs_t offset, u16 data, u16 mem_mask)
 {
-	u16 old = m_sprite_dma;
-	u16 now = COMBINE_DATA(&m_sprite_dma);
+	const u16 old = m_sprite_dma;
+	const u16 now = COMBINE_DATA(&m_sprite_dma);
 
 	if (!old && now)
 		draw_sprites(m_bitmap, m_screen->visible_area());
 }
 
 void astrocorp_state::eeprom_w(u8 data)
 {
-	ioport("EEPROMOUT")->write(data, 0xff);
+	ioport("EEPROM_OUT")->write(data, 0xff);
```

Comment:
Please use an I/O port finder for this while you’re refactoring the driver (`required_ioport` or `optional_ioport`).

Reply:
Done, for EEPROM_OUT and CPUCODE_OUT